lms/djangoapps/certificates/views/webview.py

```diff
@@ -252,7 +252,12 @@ def _update_course_context(request, context, course, course_key, platform_name):
     if CertificateGenerationCourseSetting.is_language_specific_templates_enabled_for_course(course_key):
         fields = ['start', 'end', 'max_effort', 'content_language']
         course_run_data = get_course_run_details(course_key, fields)
-        context.update(course_run_data)
+        if course_run_data['start'] and course_run_data['end'] and course_run_data['max_effort']:
+            # Calculate duration of the course run in weeks, multiplied by max_effort for total Hours of Effort
+            duration = (course_run_data['end'] - course_run_data['start']).days / 7
```

Comment:
Can we guarantee that these fields will always be a datetime object?  This seems brittle to me.  But maybe it's not

Reply:
the course run model in discovery has datetime fields for the start and end attributes, but I suppose we don't know for sure the model won't be updated. Is that what you mean?

Reply:
The JSON response is going to be a String representation of a datetime.  Unless there is some magic that is converting it to a python datetime object I would worry that we are performing these operations on strings and blowing up.